src/js/components/Calendar/Calendar.js

```diff
@@ -847,21 +847,24 @@ const Calendar = forwardRef(
         if (!children) {
           days.push(
             <CalendarDay
+              activeProp={active}
+              day={day}
               key={day.getTime()}
+              disabledProp={dayDisabled && !!dayDisabled}
               buttonProps={{
                 a11yTitle: day.toDateString(),
-                active: active && active.getTime() === day.getTime(),
-                disabled: dayDisabled && !!dayDisabled,
-                onClick: () => onClick(dateObject),
-                onMouseOver: () => setActive(dateObject),
-                onMouseOut: () => setActive(undefined),
+                onClick:
+                  dayDisabled && !!dayDisabled
```

Comment:
Shouldn't `onClick` not have an effect is the button is disabled? Seems like we're doubling up on that logic here

Reply:
I'm not setting the button to disabled but rather using aria-disabled on the button so that's why I added this check

Reply:
I see -- let's continue this thread about how to treat "disabled" in this comment thread: https://github.com/grommet/grommet/pull/7612#discussion_r2076519357